packages/api-utils/docs/generateFilterQuery.md

```diff
@@ -0,0 +1,149 @@
+
+# generateFilterQuery
+
+A function that generates a MongoDB filter query from an input GraphQL conditions object. It expects the input GraphQL conditions object to be in the format of the `FilterConditionsInput` input type (detailed below) defined in the GraphQL  Schemas along with other parameters like `context`, `collectionName` and `shopId`.
+
+As seen in the format below, 
+* The input object is a nested object where you could use  either `all` or `any` keys.
+* Both the top level `all` & `any` keys are an array of objects with one more level of `all` or `any` keys.
+* The inner `all` or `any` keys is an array of objects with the structure defined by `SingleConditionInput`.
+* The `SingleConditionInput` object has the fields which define a single condition to filter on.
+<br>
+* The `all` key is equivalent of the `$and` operator in MongoDB and the `any` key is equivalent of the `$or` operator in MongoDB.
+* In the `SingleConditionInput` object, mandatory fields are the `key`, `relationalOperator` and exactly any ONE of the value fields. The `key` is the field name to filter on. The `relationalOperator` is the relational operator to use to filter on the field (predefined as enum values). The `stringValue`, `intValue`, `floatValue`, `boolValue`, `dateValue`, `stringArrayValue`, `intArrayValue`, `floatArrayValue` are the values to filter on (use exactly one of this) depending on the key.
+* Finally there are two more optional fields `caseSensitive` and `logicalNOT`. The `caseSensitive` is a boolean flag to set if the regex is case sensitive. The `logicalNOT` is a boolean flag to set if the condition is to be negated.
+
+FilterConditionsInput format below (from GraphQL Schemas). __Example__ follows the format below:
+
+
+```graphql
+
+"Filter search with nested conditions of input (use either 'any' or 'all' not both)"
+input FilterConditionsInput {
+  "Array holding Nested conditions (use either 'any' or 'all' not both)"
```

Comment:
This seems like just asking for this to be outdated by repeating the graphql format in the doc plus I don't see how it adds a lot of value? I would just remove it

Reply:
Removed